novawallet/Modules/Staking/StakingSetupProxy/StakingConfirmProxy/StakingConfirmProxyPresenter.swift

```diff
@@ -0,0 +1,136 @@
+            balanceViewModelFactory: balanceViewModelFactory,
+            dataValidatingFactory: dataValidatingFactory,
+            localizationManager: localizationManager
+        )
+    }
+
+    override func setup() {
+        super.setup()
+
+        provideNetworkViewModel()
+        provideProxiedWalletViewModel()
+        provideProxiedAddressViewModel()
+        provideProxyAddressViewModel()
+    }
+
+    private func provideNetworkViewModel() {
+        let viewModel = networkViewModelFactory.createViewModel(from: chainAsset.chain)
+        view?.didReceiveNetwork(viewModel: viewModel)
+    }
+
+    private func provideProxiedWalletViewModel() {
+        let name = wallet.name
+
+        let icon = wallet.walletIdenticonData().flatMap { try? walletIconGenerator.generateFromAccountId($0) }
+        let iconViewModel = icon.map { DrawableIconViewModel(icon: $0) }
+        let viewModel = StackCellViewModel(details: name, imageViewModel: iconViewModel)
+        view?.didReceiveWallet(viewModel: viewModel)
+    }
+
+    private func provideProxiedAddressViewModel() {
+        guard let address = wallet.fetch(for: chainAsset.chain.accountRequest())?.toAddress() else {
+            return
+        }
+
+        let displayAddress = DisplayAddress(address: address, username: "")
+        let viewModel = displayAddressViewModelFactory.createViewModel(from: displayAddress)
+        view?.didReceiveProxiedAddress(viewModel: viewModel)
+    }
+
+    private func provideProxyAddressViewModel() {
+        let displayAddress = DisplayAddress(address: proxyAddress, username: "")
+        let viewModel = displayAddressViewModelFactory.createViewModel(from: displayAddress)
+        view?.didReceiveProxyAddress(viewModel: viewModel)
+    }
+}
+
+extension StakingConfirmProxyPresenter: StakingConfirmProxyPresenterProtocol {
+    func showProxiedAddressOptions() {
+        guard let view = view else {
+            return
+        }
+        wireframe.presentAccountOptions(
+            from: view,
+            address: "",
+            chain: chainAsset.chain,
+            locale: selectedLocale
+        )
+    }
+
+    func showProxyAddressOptions() {
+        guard let view = view else {
+            return
+        }
+        wireframe.presentAccountOptions(
+            from: view,
+            address: proxyAddress,
+            chain: chainAsset.chain,
+            locale: selectedLocale
+        )
+    }
+
+    func confirm() {
+        view?.didStartLoading()
+        interactor.submit()
+    }
+}
+
+extension StakingConfirmProxyPresenter: StakingConfirmProxyInteractorOutputProtocol {
+    func didSubmit() {
+        view?.didStopLoading()
+
+        wireframe.complete(from: view)
+    }
+
+    func didReceive(error: StakingConfirmProxyError) {
+        view?.didStopLoading()
+
+        switch error {
+        case .submit:
+            wireframe.presentRequestStatus(on: view, locale: selectedLocale) { [weak self] in
```

Comment:
We need to handle the submission errors properly because they can contain specific errors that we want to handle differently, like watch only account signing.